light/postprocess.go

```diff
@@ -380,10 +437,54 @@ func (b *BloomTrieIndexerBackend) Commit() error {
 	if err != nil {
 		return err
 	}
-	b.triedb.Commit(root, false)
-
+	var callback func(common.Hash)
+	if !b.disablePruning {
+		b.trieset.Clear()
+		callback = func(hash common.Hash) { b.trieset.Add(hash) }
+	}
+	b.triedb.Commit(root, false, callback)
+
+	// Pruning historical trie nodes if necessary.
+	if !b.disablePruning {
```

Comment:
It might be easier on the eyes if written in two branches
```golang

if b.disablePruning{
	b.triedb.Commit(root, false,nil)
}else{
	b.trieset.Clear()
	b.triedb.Commit(root, false, func(hash common.Hash) { b.trieset.Add(hash) })
	...
```